src/components/GroceryItem.svelte

```diff
@@ -0,0 +1,155 @@
+<script>
+  export let item;
+  export let removeItem;
+  export let updateItem;
+  export let editItem;
+  export let itemId;
+  export let isEditing = false;
+
+  export let inputEl;
+  let trashIconImage = "./images/trash.svg";
+  let editIconImage = "./images/edit-icon.svg";
+  //export let inputEl = item.name;
+</script>
+
+<li class="item">
+  {#if isEditing}
+    <input class="input-update" type="text" bind:value={inputEl} />
+    <button
+      class="update-btn"
+      on:click={() => {
+        updateItem(itemId, inputEl);
+      }}>Update</button
+    >
+  {:else}
+    <!--<span class="name">{item.name}</span>
+  <button class="edit-item-btn btn-white btn-animate"
+  on:click={() => {
+      isEditing=true;
+  }}><img src={editIconImage} alt="{editIconImage} image" /></button>-->
+
+    <span class="name">{item.name}</span>
+    <button
+      class="edit-item-btn btn-white btn-animate"
+      on:click={() => {
+        editItem();
+      }}><img src={editIconImage} alt="{editIconImage} image" /></button
+    >
+  {/if}
+
+  <button class="remove-item-btn" on:click={() => removeItem(itemId)}
+    ><img src={trashIconImage} alt="{trashIconImage} image" /></button
+  >
+  <!--<button class="remove-item-btn" on:click={() => removeItem(item.id)}
+    ><img src={trashIconImage} alt="{trashIconImage} image" /></button
+  >-->
+</li>
+
+<style>
+  .remove-item-btn {
+    cursor: pointer;
+    border-radius: 50%;
+    border: 1px solid var(--action-btn-color);
+    margin-top: 5px;
+  }
+  .remove-item-btn:hover {
+    background-color: var(--action-btn-color);
+    transform: translateY(-3px);
+    box-shadow: 0 10px 20px rgba(0, 0, 0, 0.2);
+  }
+
+  .name {
+    flex-grow: 1;
+    margin-right: 10px;
+  }
+
+  .item {
+    display: flex;
+    align-items: center;
+    background-color: var(--background-color-light);
+    margin-top: 5px;
+    border-radius: 10px;
+    color: var(--text-color);
+    font-weight: bold;
+    border: 1px solid var(--item-background-color);
+  }
+
+  .input-update {
+    flex-grow: 1;
+    border: none;
+    background-color: var(--background-color-light);
+    color: var(--text-color);
+    font-weight: bold;
+    text-align: center;
+    padding: 0;
+    margin: 0;
+  }
+
+  :focus {
+    outline-width: 0px;
```

Comment:
I have a challenge for you: Try to add a item to the grocery using your keyboard only. No mouse. Use the keys `Tab` and `Shift + Tab` to navigate.

You won't be able to do that because you are removing the focus styles. With this you are limiting those people who use a keyboard, from using your website. Never, ever do that. Please fix it and remember keyboard navigation, always, in your future projects.

Reference about focus styles:
- https://css-tricks.com/focusing-on-focus-styles/
- https://darekkay.com/blog/accessible-focus-indicator/

Reply:
@sandrina-p Thanks for that info. I wouldn't ever think about that by myself!

Reply:
Great ressources. Thanks.